mdata/write_buffer.go

```diff
@@ -0,0 +1,89 @@
+package mdata
+
+import (
+	"gopkg.in/raintank/schema.v1"
+)
+
+/*
+ * The write buffer keeps a window of data during which it is ok to send data out of order.
+ * Once the reorder window has passed it will try to flush the data out.
+ * The write buffer itself is not thread safe because it is only used by AggMetric,
+ * which is thread safe, so there is no locking in the buffer.
+ */
+
+type WriteBuffer struct {
```

Comment:
I find this name a bit too ambiguous, especially since the options are called "reorder-foo", so ReorderBuffer is a lot more clear IMHO.  (BTW if you use `vim-go` check out go-rename, well check out go rename either way)